src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/CompilerTypeSystemContext.BoxedTypes.cs

```diff
@@ -447,6 +447,7 @@ public override MethodIL EmitIL()
 
                 // Call an instance method on the target valuetype that has a fake instantiation parameter
                 // in it's signature. This will be swapped by the actual instance method after codegen is done.
+                codeStream.Emit(ILOpcode.tail);
```

Comment:
This is ok for now, but if we ever get slow tailcalls implemented, this won't work well anymore.